chain-vote/src/tally.rs

```diff
@@ -357,6 +389,47 @@ impl Tally {
     }
 }
 
+pub fn batch_decrypt(validated_tallies: &[ValidatedTally]) -> Result<Vec<Tally>, TallyError> {
+    let absolute_max_stake = validated_tallies.iter().map(|tally| tally.max_stake).max();
+
+    let absolute_max_stake = match absolute_max_stake {
+        Some(stake) => stake,
+        None => return Ok(vec![]),
+    };
+
+    match absolute_max_stake.try_into() {
+        // if absolute_max_stake == 0, all stakes are zero, so safe to do a trivial conversion
+        Err(_) => Ok(validated_tallies.iter().map(trivial_convert).collect()),
+        Ok(absolute_max_stake) => {
+            let table = TallyOptimizationTable::generate_with_balance(
+                absolute_max_stake,
+                1.try_into().unwrap(),
+            );
+
+            let mut result = Vec::with_capacity(validated_tallies.len());
+
+            for validated_tally in validated_tallies {
+                let tally = match validated_tally.max_stake.try_into() {
```

Comment:
nit: by relaxing unnecessary bounds in the `decrypt_tally` function we could rewrite this without the need for this match arm

Reply:
Looks like this would end up being, `validated_tally.decrypt_tally(&table, validated_tally.max_stake)`. Is it worth moving the second parameter inside the function body as `self.max_stake`?

Reply:
yup, good idea